torch/csrc/jit/graph_executor.cpp

```diff
@@ -430,6 +443,12 @@ struct GraphExecutorImpl {
   // symbolically differentiable.
   bool symbolically_differentiable;
 
+  // some ops, including python operations, can intorduce requires_grad=True
+  // variables even though no inputs to this graph are availiable, if
+  // the graph includes those operators then needGradient must be true
+  // regardles of input state.
+  bool may_introduce_gradient;
```

Comment:
I agree, these things are getting added because of actual bugs.  This one is because we changed an invariant we used to have: that the python code we call is functional. Now python code we call may be a call to a module forward with some hidden module weights that do require grad. This is pretty common when the input tensor does not requires grad but all the weights do.